plaso/parsers/ips_parser.py

```
@@ -0,0 +1,165 @@
+
+  _plugin_classes = {}
+
+  def _ParseFileEntryWithPlugin(
+      self, parser_mediator, plugin, ips_file, display_name):
+    """Parses an IPS log  file entry with a specific plugin.
+
+    Args:
+      parser_mediator (ParserMediator): parser mediator.
+      plugin (IPSPlugin): IPS parser plugin.
+      ips_file (IPSFile): IPS file.
+      display_name (str): display name.
+    """
+    required_keys_exists = plugin.CheckRequiredKeys(ips_file)
+
+    if not required_keys_exists:
+      logger.debug('Skipped parsing file: {0:s} with plugin: {1:s}'.format(
+          display_name, plugin.NAME))
+      return
+
+    logger.debug('Parsing file: {0:s} with plugin: {1:s}'.format(
+        display_name, plugin.NAME))
+
+    try:
+      plugin.UpdateChainAndProcess(parser_mediator, ips_file=ips_file)
+
+    except Exception as exception:  # pylint: disable=broad-except
+      parser_mediator.ProduceExtractionWarning((
+          'plugin: {0:s} unable to parse IPS file with error: '
+          '{1!s}').format(plugin.NAME, exception))
+
+  def ParseFileEntry(self, parser_mediator, file_entry):
+    """Parses an IPS formatted log file entry.
+
+    Args:
+      parser_mediator (ParserMediator): parser mediator.
+      file_entry (dfvfs.FileEntry): file entry to be parsed.
+    """
+    ips_file = IPSFile()
+    file_object = file_entry.GetFileObject()
+
+    display_name = parser_mediator.GetDisplayName()
+
+    try:
+      ips_file.Open(file_object)
+
+    except ValueError as exception:
+      parser_mediator.ProduceExtractionWarning(
+          '[{0:s}]unable to parse file: {1:s} with error: {2!s}'.format(
+              self.NAME, display_name, exception))
+      return
+
+    if ips_file.header and ips_file.content:
+      try:
+        for plugin in self._plugins:
+          self._ParseFileEntryWithPlugin(
+              parser_mediator, plugin, ips_file, display_name)
+
+      except Exception as exception:  # pylint: disable=broad-except
```

@rick-slin why the double catch all Exception ?

---

That is an oversight on my part. I can see that the one here is redundant with the one in _ParseFileEntryWithPlugin

---

That is an oversight on my part. I can see that the one here is redundant with the one in _ParseFileEntryWithPlugin.